checkov/terraform/checks/resource/aws/SNSTopicPolicyAnyPrincipal.py

```diff
@@ -20,9 +20,12 @@ def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
         conf_policy = conf.get("policy")
         if conf_policy:
             if isinstance(conf_policy[0], dict):
-                policy = Policy(conf["policy"][0])
-                if policy.is_internet_accessible():
-                    return CheckResult.FAILED
+                try:
+                    policy = Policy(conf["policy"][0])
+                    if policy.is_internet_accessible():
+                        return CheckResult.FAILED
+                except AttributeError:
+                    return CheckResult.UNKNOWN
```

Comment:
@ChanochShayner This is pretty wide except... Can you make it narrower, or add a UT and log the error at least?

Reply:
Agree + UT should be added anyways please